astropy/wcs/wcsapi/high_level_api.py

```diff
@@ -240,6 +242,17 @@ def high_level_objects_to_values(*world_objects, low_level_wcs):
         else:
             world.append(rec_getattr(objects[key], attr))
 
+    # Check the type of the return values - should be scalars or plain Numpy
+    # arrays, not e.g. Quantity. Note that we deliberately use type(w) because
+    # we don't want to match Numpy subclasses.
+    for w in world:
```

Comment:
Will this have performance impact? 🤔

Reply:
I very much doubt it, it's only looping over a couple (or maybe up to 5) dimensions - not over each individual value of a large array